dynamic.js

```diff
@@ -3,16 +3,31 @@
 const fs = require('fs')
 const path = require('path')
 const yaml = require('js-yaml')
+const Ref = require('json-schema-resolver')
 
 module.exports = function (fastify, opts, next) {
   fastify.decorate('swagger', swagger)
 
   const routes = []
+  const sharedSchemasMap = new Map()
+  let ref
 
   fastify.addHook('onRoute', (routeOptions) => {
     routes.push(routeOptions)
   })
 
+  fastify.addHook('onRegister', async (instance) => {
+    instance.addHook('onReady', (done) => {
+      const allSchemas = instance.getSchemas()
+      for (const schemaId of Object.keys(allSchemas)) {
+        if (!sharedSchemasMap.has(schemaId)) {
```

Comment:
Can we detect if the id is being overridden by something else?

Reply:
Maybe here we can add instance prefix to schema name if it exists? In some cases I guess it could solve some schema conflicts.

Reply:
The problem here is that onRegister is receiving for all the register so in this case:

- register root - addSchema 1
  - child 1 registered - addSchema 2
     - subchild 2 registered - addSchema 3

For subchild 2 i will read schema 1, 2, 3
For child 1 schema 1 and 2
For root only schema 1

So it will be quite normal to have duplicate and I think it is ok since the `addSchema` already has a check to verify duplicate `$id`

Reply:
I mean slightly different case.

* root - schema-1
  * child1 (prefix '/news') - item
  * child2 (prefix '/comments') - item

So if `item` schemas are registered in isolated scopes they will know now nothing about each other. However when you gather all them in shared schema map they will have the same id.

Reply:
Could you leave a TODO comment in here? Or open an issue? I think we might want/be able to have a way to get _all schemas defined at the current level_ so you do not get them all here.

Reply:
I agree and I will describe the case 👍